### PR TITLE
Bug 1228600 - Migrate SimLockSystemDialog to a WebComponent r=mhenretty

### PR DESCRIPTION
**Pull request by albertopq**

### CODE REVIEW

**Comment by autolander** [comment]
Running tests
https://treeherder.mozilla.org/#/jobs?repo=gaia&revision=31068a65e709594e240ab4c27d273a4ba48f7ee4

**Comment by autolander** [comment]
Running tests
https://treeherder.mozilla.org/#/jobs?repo=gaia&revision=54d924d4cba25a35b68575cbfce9c11d127f3ec1

**Comment by autolander** [comment]
Running tests
https://treeherder.mozilla.org/#/jobs?repo=gaia&revision=d1f327c3a7c535c77ef67accc0860ccd18abeb56

**Comment by autolander** [comment]
Running tests
https://treeherder.mozilla.org/#/jobs?repo=gaia&revision=b0aeb04d801d49f36d684c6e84f1d63077b9478b

File: apps/system/js/sim_lock_system_dialog.js [hunk 1]
```diff
@@ -123,34 +129,35 @@
...
     document.activeElement.scrollIntoView(false);
   };
 
   SimLockSystemDialog.prototype._registerEvents = function() {
-    this.dialogDone.onclick = this.verify.bind(this);
-    this.dialogSkip.onclick = this.skip.bind(this);
+    this.dialogDone.addEventListener('click', this.verify.bind(this));
+    this.dialogSkip.addEventListener('click', this.skip.bind(this));
     this.dialogDone.onmousedown = this._handle_mousedown.bind(this);
     this.dialogSkip.onmousedown = this._handle_mousedown.bind(this);
```
Comment (gmarty) [inline apps/system/js/sim_lock_system_dialog.js]: If you're changing `.onclick` to `addEventListener`, why not do the same for `.onmousedown`?

File: apps/system/js/sim_lock_system_dialog.js [hunk 2]
```diff
@@ -167,10 +174,13 @@
     this.errorMsgBody = document.getElementById('messageBody');
 
     this.containerDiv = document.querySelector('#simlock-dialog .container');
+    this.dialogDone = document.querySelector('#simlock-dialog button.submit');
+    this.dialogSkip = document.querySelector('#simlock-dialog button.cancel');
```
Comment (gmarty) [inline apps/system/js/sim_lock_system_dialog.js]: nit: I'd replace all `document.querySelector('#simlock-dialog selector')` by `this.element.querySelector('selector')`.
It's easier to read and slightly more efficient.

**Comment by autolander** [comment]
Running tests
https://treeherder.mozilla.org/#/jobs?repo=gaia&revision=151048ae4b6488404642b34cf44cff3cf1c52a99

**Comment by autolander** [comment]
Running tests
https://treeherder.mozilla.org/#/jobs?repo=gaia&revision=651e47cca329e2de5bd4ed08b31b0be2edfca806

**Pull request merged**